healthcare/api-client/v1/dicom/dicomweb.py

```diff
@@ -0,0 +1,434 @@
+# [END healthcare_dicomweb_store_instance]
+
+
+# [START healthcare_dicomweb_search_instances]
+def dicomweb_search_instance(
+    base_url, project_id, cloud_region, dataset_id, dicom_store_id
+):
+    """Handles the GET requests specified in DICOMweb standard."""
+    url = "{}/projects/{}/locations/{}".format(base_url, project_id, cloud_region)
+
+    dicomweb_path = "{}/datasets/{}/dicomStores/{}/dicomWeb/instances".format(
+        url, dataset_id, dicom_store_id
+    )
+
+    # Make an authenticated API request
+    session = get_session()
+
+    headers = {"Content-Type": "application/dicom+json; charset=utf-8"}
+
+    response = session.get(dicomweb_path, headers=headers)
+    response.raise_for_status()
+
+    instances = response.json()
+
+    print("Instances:")
+    print(json.dumps(instances, indent=2))
+
+    return instances
+
+
+# [END healthcare_dicomweb_search_instances]
+
+
+# [START healthcare_dicomweb_retrieve_study]
+def dicomweb_retrieve_study(
+    base_url, project_id, cloud_region, dataset_id, dicom_store_id, study_uid
+):
+    """Handles the GET requests specified in the DICOMweb standard."""
+    url = "{}/projects/{}/locations/{}".format(base_url, project_id, cloud_region)
+
+    dicomweb_path = "{}/datasets/{}/dicomStores/{}/dicomWeb/studies/{}".format(
+        url, dataset_id, dicom_store_id, study_uid
+    )
+
+    # When specifying the output file, use an extension like ".multipart."
+    # Then, parse the downloaded multipart file to get each individual
+    # DICOM file.
+    file_name = "study.multipart"
+
+    # Make an authenticated API request
+    session = get_session()
+
+    response = session.get(dicomweb_path)
+
+    response.raise_for_status()
+
+    with open(file_name, "wb") as f:
+        f.write(response.content)
+        print("Retrieved study and saved to {} in current directory".format(file_name))
+
+    return response
+
+
+# [END healthcare_dicomweb_retrieve_study]
+
+
+# [START healthcare_dicomweb_search_studies]
+def dicomweb_search_studies(
+    base_url, project_id, cloud_region, dataset_id, dicom_store_id
+):
+    """Handles the GET requests specified in the DICOMweb standard."""
+    url = "{}/projects/{}/locations/{}".format(base_url, project_id, cloud_region)
+
+    dicomweb_path = "{}/datasets/{}/dicomStores/{}/dicomWeb/studies".format(
+        url, dataset_id, dicom_store_id
+    )
+
+    # Refine your search by appending DICOM tags to the
+    # request in the form of query parameters. This sample
+    # searches for studies containing a patient's name.
+    params = {"PatientName": "Sally Zhang"}
+
+    session = get_session()
+
+    response = session.get(dicomweb_path, params=params)
+
+    response.raise_for_status()
+
+    print("Studies found: response is {}".format(response))
+
+    # Uncomment the following lines to process the response as JSON.
+    # patients = response.json()
+    # print('Patients found matching query:')
+    # print(json.dumps(patients, indent=2))
+
```

Comment:
Any reason why we are commenting this out? Commented lines aren't preferred since they aren't tested.

Reply:
The test medical image we use is quite bare and doesn't have a person's name in it. If we try to search for it using params like in the example with "Sally Zhang", it will always fail.